app/transport/rest/user.go

```diff
@@ -60,3 +69,40 @@ func (uh UserHandler) CreateUser() http.Handler {
 		uh.logger.Debugw("User successfully created", "ID", id)
 	})
 }
+
+// GetUserByID will handle user search.
+func (uh UserHandler) GetUserByID() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		id := mux.Vars(req)["id"]
+
+		if _, err := uuid.Parse(id); err != nil {
+			uh.logger.Warnw("Invalid UUID", "ID", id)
+			WriteJSONResponse(w, http.StatusBadRequest, Response{Message: MsgBadRequest, Details: err.Error()}, uh.logger)
+
+			return
+		}
+
+		entUser, err := uh.usecase.GetUser(id)
+		if err != nil {
+			if errors.Is(err, globals.ErrNotFound) {
+				uh.logger.Debugw("No user found.", "ID", id)
+				WriteJSONResponse(w, http.StatusNotFound, Response{Message: MsgNotFound, Details: err.Error()}, uh.logger)
```

Comment:
There's no much sense to propagate this error to the client because it will be the same for every 404 returned here, you can write a pretty message here instead. But let it be, not much harm from that.